File: src/KR.Hanyang.Mindwatch/KR.Hanyang.Mindwatch.Frontend/src/models/QuestionnaireRun.ts
import { Answer } from "./Answer";
import { Employee } from "./Employee";
import { Questionnaire } from "./Questionnaire";
import { QuestionnaireRunStatus } from "./QuestionnaireRunStatus";

export interface QuestionnaireRun {
  id?: number;
  questionnaireId: number;
  questionnaire?: Questionnaire;
  employeeId: number;
  employee?: Employee;
  questionnaireRunStatus?: QuestionnaireRunStatus;
  createdAt?: string; // ISO DateTime string
  openDateTime?: string; // ISO DateTime string
  closeDateTime?: string; // ISO DateTime string
  answers?: Answer[];
}

File: src/KR.Hanyang.Mindwatch/KR.Hanyang.Mindwatch.Domain/Entities/QuestionnaireRun.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KR.Hanyang.Mindwatch.Domain.Entities
{
    public class QuestionnaireRun
    {
        public int Id { get; set; }
        public int QuestionnaireId { get; set; }
        public Questionnaire? Questionnaire { get; set; }
        public QuestionnaireRunStatus QuestionnaireRunStatus { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime OpenDateTime { get; set; }
        public DateTime CloseDateTime { get; set; }
        public ICollection<Answer>? Answers { get; set; }
    }
}
